core/src/main/scala/org/apache/spark/sql/delta/DeltaLog.scala

```diff
@@ -192,6 +192,13 @@ class DeltaLog private(
     }
   }
 
+  /**
+   * [[IsolationLevel]] as set in table delta config and to be passed into the transaction.
+   */
+  private[delta] def isolationLevel: IsolationLevel = {
+    DeltaConfigs.ISOLATION_LEVEL.fromMetaData(metadata)
+  }
+
```

Comment:
Why do we add this to `DeltaLog` instead of just getting it directly from the metadata in `OptimisticTransaction`?

Reply:
good point. i initially put it on `DeltaLog` because i wast sure where else it was going to be used, but it really only relates to `OptimisticTransaction`. will fix.

Reply:
i did keep `DeltaLog.isolationLevel` too for tests. the alternative is to open up `OptimisticTransactoin.getDefaultIsolationLevel` so it can be inspected in test.

Reply:
hmm I think I'd prefer that than adding this just solely for testing purposes

Reply:
sure thats easy change